### PR TITLE
gh-89682: [doc] reword docstring of __contains__ to clarify that it returns a bool

### PR DESCRIPTION
**Pull request by ivanistheone**
I was looking to implement a "custom object" to represent intervals of numbers

e.g. `if x in interval(2,4)` will be `True if (x >= 2) and (x < 4)`.

I looked up `help(list.__contains__)` and it says "Return key in self" which I interpreted as returning the actual key, not a boolean. A moment of confusion ensured and I tested `[2,3].__contains__(2)` to check that return value of `__contains__` is indeed boolean.

The docstring change suggestion makes the return type of `__contains__` clearer.

A type hint for old school people who `?` methods in jupyter notebooks ;)

* Issue: gh-89682

### CODE REVIEW

**Review by MaxwellDupre — APPROVED** [review]
Looks ok.

File: Objects/typeobject.c [hunk 1]
```diff
@@ -8139,7 +8139,7 @@ static slotdef slotdefs[] = {
     SQSLOT("__delitem__", sq_ass_item, slot_sq_ass_item, wrap_sq_delitem,
            "__delitem__($self, key, /)\n--\n\nDelete self[key]."),
     SQSLOT("__contains__", sq_contains, slot_sq_contains, wrap_objobjproc,
-           "__contains__($self, key, /)\n--\n\nReturn key in self."),
+           "__contains__($self, key, /)\n--\n\nReturn True if key in self."),
```
Comment (iritkatriel) [inline Objects/typeobject.c]:
```suggestion
           "__contains__($self, key, /)\n--\n\nReturn bool(key in self)."),
```

**Comment by iritkatriel** [comment]
We'll need a news file for this. It's minor, but it touches a builtin.

File: Misc/NEWS.d/next/Documentation/2022-11-26-21-43-05.gh-issue-89682.DhKoTM.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Reworded docstring of ``__contains__`` to clarify that it returns a ``bool``.
```
Comment (iritkatriel) [inline Misc/NEWS.d/next/Documentation/2022-11-26-21-43-05.gh-issue-89682.DhKoTM.rst]:
```suggestion
Reworded docstring of the default ``__contains__`` to clarify that it returns a :class:`bool`.
```

**Comment by iritkatriel** [comment]
Thanks, I'll merge once CI completes.

**Pull request merged**